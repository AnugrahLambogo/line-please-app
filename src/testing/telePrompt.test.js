import React from 'react';
import ReactDOM from 'react-dom'
import TelePrompt from '../components/teleprompt'
import renderer from 'react-test-renderer'

describe('TelePrompt component', () =>{
it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<TelePrompt />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('TelePrompt renders the UI as expected', () => {
    const tree = renderer
                .create(<TelePrompt />)
                .toJSON();
    expect(tree).toMatchSnapshot();
});

})