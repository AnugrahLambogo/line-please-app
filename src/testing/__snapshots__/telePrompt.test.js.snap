// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TelePrompt component TelePrompt renders the UI as expected 1`] = `
<section>
  <h2
    id="currentText"
  >
    
  </h2>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="chooseText"
    >
      Switch text:
    </label>
    <select
      id="chooseText"
      name="chooseText"
      onChange={[Function]}
    >
      <option>
        -Select Text-
      </option>
    </select>
    <button
      type="submit"
    >
      Switch!
    </button>
  </form>
  <form>
    <label
      htmlFor="lineOrSection"
    >
      Display by:
    </label>
    <select
      id="lineOrSection"
      name="lineOrSection"
      onChange={[Function]}
    >
      <option>
        Section
      </option>
      <option>
        Line
      </option>
    </select>
  </form>
  <form>
    <label
      htmlFor="displayTime"
    >
      How many seconds to display:
    </label>
    <input
      defaultValue="3"
      id="displayTime"
      name="displayTime"
      onChange={[Function]}
      type="number"
    />
  </form>
  <form>
    <select
      id="fullOrEmpty"
      name="fullOrEmpty"
      onChange={[Function]}
    >
      <option>
        Show full line
      </option>
      <option>
        Show only the beginning
      </option>
    </select>
    <button
      id="startButton"
      onClick={[Function]}
    >
      Start/Pause
    </button>
  </form>
  <section
    className="display"
  >
    <ul
      className="orationDisplay"
    >
      <li
        className="line"
      >
         
        <p>
          Select which text you want to practice!
        </p>
         
      </li>
    </ul>
  </section>
  <section>
    <button
      className="teleButton"
      onClick={[Function]}
    >
      back
    </button>
    <button
      className="teleButton"
      onClick={[Function]}
    >
      next
    </button>
    <button
      className="teleButton"
      onClick={[Function]}
    >
      Restart
    </button>
  </section>
     
</section>
`;
